spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java

```diff
@@ -68,12 +102,34 @@ private static boolean isMutable(JavaClass cls) {
     }
 
     public static boolean looksLikeASetter(Method method, JavaClass cls) {
+        // Static methods are no setters.
+        if (method.isStatic()) {
+            return false;
+        }
+
+        // Non-public setters cannot be used from outside the package so disregard them.
```

Comment:
I think this change is not so natural. The package affect nothing to the immutability of class architecture.

Reply:
Yes, this can be disputed. If you whish, I can change this part to only omit methods which are private or protected.